lib/puppet/provider/mount/parsed.rb

```diff
@@ -264,7 +272,12 @@ def self.mountinstances
     end
     mount_output.each do |line|
       if match = regex.match(line) and name = match.captures.first
-        instances << {:name => name, :mounted => :yes} # Only :name is important here
+        if Facter.value(:kernel) == 'Linux'
```

Comment:
Bringing in @branan for feedback on this one. I wasn't quite sure what conclusion we came to, if we should include that work in this ticket or file a new one. It seems like Darwin and SunOS at least would be pretty trivial to add support for but I'm not sure about anything else